src/views/ApartmentApp/components/StudentApplication/components/BottomBar/index.js

```diff
@@ -0,0 +1,136 @@
+import SaveButton from './components/SaveButton';
+
+const submitAlertText = (
+  <span>
+    This feature is not yet implemented.
+    <br />
+    Clicking the "Accept" button will simply hide the application cards.
+  </span>
+);
+
+const BottomBar = ({
+  applicationCardsOpen,
+  applicationID,
+  canEditApplication,
+  disableSubmit,
+  editorUsername,
+  saving,
+  submitDialogOpen,
+  unsavedChanges,
+  onCloseDialog,
+  onCloseOkay,
+  onSaveButtonClick,
+  onShowApplication,
+  onSubmitAppAccepted,
+  onSubmitButtonClick,
+}) => {
+  const dynamicContent = !applicationCardsOpen
+    ? !applicationID
+      ? {
+          primaryText: 'You are not on any applications for the current semester.',
+          secondaryText: 'Click the button to begin a new application.',
+          buttonLabel: 'Create a new application',
+        }
+      : canEditApplication
+      ? {
+          primaryText: 'You are on an existing application for this semester.',
+          secondaryText: "Click the button to edit your group's application.",
+          buttonLabel: 'Edit your application',
+        }
+      : {
+          primaryText:
+            'You are on an existing application for this semester, but you are not assigned as the editor.',
+          secondaryText: 'Only the application editor may edit the application.',
+          buttonLabel: 'View your application',
+        }
+    : saving === 'failed'
+    ? {
+        primaryText: 'Something went wrong while trying to save the application.',
+        secondaryText: null,
+        itemProps: {
+          variant: 'overline',
+          color: 'error',
+        },
+      }
+    : unsavedChanges
+    ? {
+        primaryText: 'You have unsaved changes.',
+        secondaryText: null,
+      }
+    : {
+        primaryText: 'All changes to be saved.',
+        secondaryText: null,
+      };
+
+  return (
+    <Card className={'sticky-page-bottom-bar'} variant="outlined">
+      <CardContent>
+        <Grid container direction="row" justify="flex-end" spacing={2}>
+          <Grid item xs={12} sm>
+            <Typography variant="body1" {...dynamicContent.itemProps}>
+              {dynamicContent.primaryText}
+              {dynamicContent.secondaryText && <br />}
+              {dynamicContent.secondaryText}
+            </Typography>
+          </Grid>
+          {!applicationCardsOpen ? (
+            <Grid item xs={12} sm={4}>
+              <Button
+                variant="contained"
+                onClick={onShowApplication}
+                color="secondary"
+                fullWidth
+                disabled={applicationCardsOpen}
+              >
+                {dynamicContent.buttonLabel}
+              </Button>
+            </Grid>
+          ) : (
+            <Grid container item xs={12} sm={6} lg={4} spacing={2}>
+              <Grid item xs>
+                <SaveButton
+                  saving={saving}
+                  onClick={canEditApplication && onSaveButtonClick}
```

Comment:
I figured if it would for abstract objects, JSX components, and every other thing I have tried so far, then it would probably work for this too, so I tried it out. My original plan was a `canEditApplication ? onSaveButtonClick : null` which I knew was valid syntax with functions from past things and examples in documentation, so it seemed like a reasonable guess

Reply:
Right, that makes sense because of course `&&` is a Logical AND, not a Boolean AND in JavaScript, so it's value resolves to the first falsy value or the last value. That is, when `canEditApplication` is false, `&&` returns false, but when `canEditApplication` is true, `&&` will evaluate onSaveButtonClick (which is a truthy value) and return that. If at some point `canEditApplication` were `undefined`, then `&&` would actually return `undefined` which would be fine in this case, but might be unexpected for people like me who think of it as a Boolean operator rather than a 'Logical Operator'.